src/OrchardCore.Modules/OrchardCore.Microsoft.Authentication/Configuration/OpenIdConnectOptionsConfiguration.cs

```diff
@@ -38,7 +39,7 @@ public void Configure(string name, OpenIdConnectOptions options)
             options.SignInScheme = "Identity.External";
             options.UseTokenLifetime = true;
             options.SaveTokens = _azureADSettings.SaveTokens;
-
+            options.TokenValidationParameters.IssuerValidator = AadIssuerValidator.GetAadIssuerValidator(options.Authority, options.Backchannel).Validate;
```

Comment:
If a tenant ID is specified, the behavior is the same as with the default issuer validation used until now. 
In case of 'common' or 'organizations', the AadIssuerValidator basically replaces the tenant ID placeholder in the multi-tenant metadata before checking it. For details, see https://learn.microsoft.com/en-us/entra/identity-platform/access-tokens#validate-the-issuer